src/message_interface/mod.rs

```diff
@@ -126,9 +129,114 @@ mod tests {
 
     const TOKEN_SUPPLY: u64 = 1_813_620_509_061_365;
 
+    #[tokio::test]
+    #[cfg(feature = "events")]
+    async fn message_interface_emit_event() {
+        let storage_path = "test-storage/message_interface_emit_event";
+        std::fs::remove_dir_all(storage_path).unwrap_or(());
```

Comment:
setup ?

Reply:
Same for the whole file

Reply:
dont think we can refer to tests/common from inside the crate right?

Reply:
Ah yeah, fair enough 🤦🏻